MdePkg/Library/BasePeCoffLib/BasePeCoff.c

```diff
@@ -1054,15 +1056,17 @@ PeCoffLoaderRelocateImage (
     RelocDir = &Hdr.Te->DataDirectory[0];
   }
 
-  if ((RelocDir != NULL) && (RelocDir->Size > 0) && ((RelocDir->Size - 1) < (MAX_UINT32 - RelocDir->VirtualAddress))) {
+  Status = SafeUint32Add (RelocDir->VirtualAddress, (RelocDir->Size - 1), &EndAddress);
+  if ((RelocDir != NULL) && (RelocDir->Size > 0) && (!RETURN_ERROR (Status))) {

```

Comment:
I think the SafeUint32Add() operation should be moved within this if() statement and check for error status in a nested if.  Same comment applies to change below